frontend/src/logger/pcm.logger.ts

```diff
@@ -0,0 +1,75 @@
+import {AxiosInstance} from 'axios'
+import {AppConfig} from '../app-config/types'
+import {handleNotAuthorizedErrors} from '../auth/handleNotAuthorizedErrors'
+import identityFn from 'lodash/identity'
+
+enum LogLevel {
+  info = 'INFO',
+  warning = 'WARNING',
+  debug = 'DEBUG',
+  error = 'ERROR',
+  critical = 'CRITICAL',
+}
+
+interface LogEntry {
+  message: string
+  level: LogLevel
+  extra?: {[key: string]: string}
+}
+
+export class PcmLogger {
```

Comment:
question: Have you thought of implementing the same interface (e.g. `ILogger`) with a `ConsoleLogger` to be used in local development? Not sure about this one, I'd leave it up to you to consider whether this is needed

Reply:
Yes I thought about it, but since we always run the backend also in dev mode, logs reach the backend and can be seen there, I don't think it would bring much value, but rather making logs easier to see
I'm open to it!